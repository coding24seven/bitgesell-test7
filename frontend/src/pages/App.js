import React from 'react';
import { Routes, Route, Link } from 'react-router-dom';
import Items from './Items';
import ItemDetail from './ItemDetail';
import { DataProvider } from '../state/DataContext';

function App() {
  return (
    <DataProvider>
      <nav style={{padding: 16, borderBottom: '1px solid #ddd'}}>
        <Link to="/">Items</Link>
        <br/>
        <Link to="/unmount">Unmount Items.js while items are loading</Link>
      </nav>
      <Routes>
        <Route path="/" element={<Items />} />
        <Route path="/items/:id" element={<ItemDetail />} />
        <Route path="/unmount" element={<h1>Items.js unmounted</h1>} />
      </Routes>
    </DataProvider>
  );
}

export default App;