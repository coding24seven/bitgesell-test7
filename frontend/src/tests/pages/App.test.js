import React from 'react';
import { render } from '@testing-library/react';
import App from '../../pages/App';
import { BrowserRouter } from 'react-router-dom';

describe('App', () => {
  test('renders initial content', () => {
    const { container } = render(<BrowserRouter><App/></BrowserRouter>);

    expect(container).toMatchSnapshot();
  });
})
