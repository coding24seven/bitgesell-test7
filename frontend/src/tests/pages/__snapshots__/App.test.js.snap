// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders initial content 1`] = `
<div>
  <nav
    style="padding: 16px; border-bottom: 1px solid #ddd;"
  >
    <a
      href="/"
    >
      Items
    </a>
    <br />
    <a
      href="/unmount"
    >
      Unmount Items.js while items are loading
    </a>
  </nav>
  <input />
  <button>
    Search
  </button>
  <label>
    Page
    <input
      max="0"
      min="1"
      style="width: 60px;"
      type="number"
      value="1"
    />
  </label>
  <p>
    Loading...
  </p>
</div>
`;

exports[`renders App component 1`] = `
<div>
  <nav
    style="padding: 16px; border-bottom: 1px solid #ddd;"
  >
    <a
      href="/"
    >
      Items
    </a>
    <br />
    <a
      href="/unmount"
    >
      Unmount Items.js while items are loading
    </a>
  </nav>
  <input />
  <button>
    Search
  </button>
  <label>
    Page
    <input
      max="0"
      min="1"
      style="width: 60px;"
      type="number"
      value="1"
    />
  </label>
  <p>
    Loading...
  </p>
</div>
`;
